python_bindings/dda_model/optimizers.py

```diff
@@ -16,8 +16,20 @@ def __call__(self, gradients):
         self.S = self.beta2 * self.S + (1 - self.beta2) * (np.power(gradients, 2)) / (1 - self.beta2**(self.timestep + 1))
 
         for i in range(len(gradients)):
-            result[i] = self.V[i] / (np.sqrt(self.S[i]) + 0.00000001)
+            result[i] = self.V[i] / (np.sqrt(self.S[i]) + 1e-8)
         
         self.timestep += 1
 
+        return result
+    
+class AdaGradOptimizer:
+    def __init__(self):
+        self.squaredGradient = 0
+
+    def __call__(self, gradients):
+
+        result = np.array(gradients)
+        print("Using Adagrad Optimizer.")
+        self.squaredGradient += np.power(gradients, 2) / 100000
```

Comment:
Why divided by 100k?

Reply:
I am not sure...this is what Yage had in his implementation but it may not be correct (as his things usually are)

Reply:
My implementation does not look like the link that you sent. might need to take a look at this. I will leave a comment in my code